VM build failure - VMState: 0x00050080 or 0x000501ff

Got this on https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-mac-x64-openj9/573 (`build-macstadium-macos1014-1`)
```
* For target support_jmods_java.compiler.jmod:
Assertion failed at /Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/build/macosx-x86_64-normal-server-release/vm/compiler/../compiler/compile/J9SymbolReferenceTable.cpp:811: containingClass != NULL
VMState: 0x00050080
	failed to get defining class of field ref cpIndex=35 in owning method J9Method=0xc5e3660
compiling jdk/internal/org/objectweb/asm/ClassWriter.<init>(I)V at level: warm

JVMDUMP039I Processing dump event "abort", detail "" at 2020/04/30 00:59:05 - please wait.
JVMDUMP032I JVM requested System dump using '/Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/make/core.20200430.005905.52000.0001.dmp' in response to an event
JVMDUMP012E Error in System dump: The core file created by child process with pid = 52079 was not found. Expected to find core file with name "/cores/core.52079"
JVMDUMP032I JVM requested Java dump using '/Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/make/javacore.20200430.005905.52000.0002.txt' in response to an event
JVMDUMP010I Java dump written to /Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/make/javacore.20200430.005905.52000.0002.txt
JVMDUMP032I JVM requested Snap dump using '/Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/make/Snap.20200430.005905.52000.0003.trc' in response to an event
JVMDUMP010I Snap dump written to /Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/make/Snap.20200430.005905.52000.0003.trc
JVMDUMP007I JVM Requesting JIT dump using '/Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/make/jitdump.20200430.005905.52000.0004.dmp'
JVMDUMP010I JIT dump written to /Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/make/jitdump.20200430.005905.52000.0004.dmp
JVMDUMP013I Processed dump event "abort", detail "".

* All command lines available in /Users/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-mac-x64-openj9/workspace/build/src/build/macosx-x86_64-normal-server-release/make-support/failure-logs.
```
Access to the machine can be granted by opening an issue at https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues

New occurrence: https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-linux-ppc64le-openj9-linuxXL/413/console
```
00:11:29  Assertion failed at ../../../../../openj9/runtime/compiler/compile/J9SymbolReferenceTable.cpp:872: containingClass != NULL
00:11:29  VMState: 0x00050080
00:11:29  	failed to get defining class of field ref cpIndex=3 in owning method J9Method=0x100004321780
00:11:29  compiling java/util/Spliterators$IteratorSpliterator.estimateSize()J at level: cold
00:11:29  #0: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0xad6aa0) [0x100001c76aa0]
00:11:29  #1: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0xae725c) [0x100001c8725c]
00:11:29  #2: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x6f4348) [0x100001894348]
00:11:29  #3: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x6f62dc) [0x1000018962dc]
00:11:29  #4: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x182b18) [0x100001322b18]
00:11:29  #5: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x3402cc) [0x1000014e02cc]
00:11:29  #6: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x343d2c) [0x1000014e3d2c]
00:11:29  #7: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x316ee0) [0x1000014b6ee0]
00:11:29  #8: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x318afc) [0x1000014b8afc]
00:11:29  #9: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x3191d4) [0x1000014b91d4]
00:11:29  #10: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x6b95bc) [0x1000018595bc]
00:11:29  #11: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x693d78) [0x100001833d78]
00:11:29  #12: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x1aab28) [0x10000134ab28]
00:11:29  #13: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x1abddc) [0x10000134bddc]
00:11:29  #14: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9prt29.so(+0x3d508) [0x100000dcd508]
00:11:29  #15: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x1ae1ec) [0x10000134e1ec]
00:11:29  #16: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x1ae94c) [0x10000134e94c]
00:11:29  #17: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x1a8ff4) [0x100001348ff4]
00:11:29  #18: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x1a9710) [0x100001349710]
00:11:29  #19: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x1a9818) [0x100001349818]
00:11:29  #20: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9prt29.so(+0x3d508) [0x100000dcd508]
00:11:29  #21: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9jit29.so(+0x1a9e64) [0x100001349e64]
00:11:29  #22: /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/jdk/lib/default/libj9thr29.so(+0x11268) [0x100000e61268]
00:11:29  #23: /lib64/libpthread.so.0(+0x8cd4) [0x1000000c8cd4]
00:11:29  #24: function clone+0xe4 [0x100000297f14]
00:11:29  
00:11:29  JVMDUMP039I Processing dump event "abort", detail "" at 2020/11/24 00:11:28 - please wait.
00:11:29  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/make/core.20201124.001128.24377.0001.dmp' in response to an event
00:11:38  JVMDUMP010I System dump written to /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/make/core.20201124.001128.24377.0001.dmp
00:11:38  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/make/javacore.20201124.001128.24377.0002.txt' in response to an event
00:11:40  JVMDUMP010I Java dump written to /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/make/javacore.20201124.001128.24377.0002.txt
00:11:40  JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/make/Snap.20201124.001128.24377.0003.trc' in response to an event
00:11:40  JVMDUMP010I Snap dump written to /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/make/Snap.20201124.001128.24377.0003.trc
00:11:40  JVMDUMP032I JVM requested JIT dump using '/home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/make/jitdump.20201124.001128.24377.0004.dmp' in response to an event
00:11:41  JVMDUMP010I JIT dump written to /home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/make/jitdump.20201124.001128.24377.0004.dmp
00:11:41  JVMDUMP013I Processed dump event "abort", detail "".
00:11:42  CreateJmods.gmk:217: recipe for target '/home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/images/jmods/java.instrument.jmod' failed
00:11:42  gmake[3]: *** [/home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-ppc64le-openj9-linuxXL/workspace/build/src/build/linux-ppc64le-normal-server-release/images/jmods/java.instrument.jmod] Error 1
00:11:42  make/Main.gmk:357: recipe for target 'java.instrument-jmod' failed
00:11:42  gmake[2]: *** [java.instrument-jmod] Error 1
```

--- a/runtime/compiler/compile/J9Compilation.cpp
+++ b/runtime/compiler/compile/J9Compilation.cpp
@@ -189,12 +189,7 @@ J9::Compilation::Compilation(int32_t id,
    _perClientMemory(_trMemory),
    _methodsRequiringTrampolines(getTypedAllocator<TR_OpaqueMethodBlock *>(self()->allocator())),
 #endif /* defined(J9VM_OPT_JITSERVER) */
-   _osrProhibitedOverRangeOfTrees(false),
-   _useTracingBuffer(false),
-   _tracingBufferStart(NULL),
-   _tracingBufferCursor(NULL),
-   _tracingBufferSize(0),
-   _tracingBufferFreeSpace(0)
+   _osrProhibitedOverRangeOfTrees(false)
    {
    _symbolValidationManager = new (self()->region()) TR::SymbolValidationManager(self()->region(), compilee);
 
@@ -236,14 +231,6 @@ J9::Compilation::Compilation(int32_t id,
 J9::Compilation::~Compilation()
    {
    _profileInfo->~TR_AccessedProfileInfo();
-   }
-
-void
-J9::Compilation::allocateTracingBuffer()
-   {
-   const int32_t size = TRACING_BUFFER_CAPACITY;
-   _tracingBufferStart = (char *)self()->region().allocate(size);
-   _tracingBufferSize = size;
    }
 
 TR_J9VMBase *

--- a/runtime/compiler/compile/J9Compilation.hpp
+++ b/runtime/compiler/compile/J9Compilation.hpp
@@ -75,22 +75,6 @@ class ClientSessionData;
 #define COMPILATION_RESERVE_NTRAMPOLINES_ERROR_INBINARYENCODING -19
 #define COMPILATION_AOT_RELOCATION_FAILED -20
 
-#define TRACING_BUFFER_CAPACITY 10*1024
-#define TRACING_BUFFER_SEGMENT_SIZE 128
-#define ADD_TRACING_BUFFER_MESSAGE(comp, fmt, ...) \
-   do \
-      { \
-      if (comp->useTracingBuffer()) \
-         { \
-         char *buffer = comp->getTracingBufferCursor(); \
-         if (comp->getTracingBufferFreeSpace() >= TRACING_BUFFER_SEGMENT_SIZE) \
-            { \
-            snprintf(buffer, TRACING_BUFFER_SEGMENT_SIZE, fmt, ##__VA_ARGS__); \
-            comp->incTracingBuffer(TRACING_BUFFER_SEGMENT_SIZE); \
-            } \
-         } \
-      } while (false)
-
 
 
 namespace J9
@@ -367,23 +351,6 @@ class OMR_EXTENSIBLE Compilation : public OMR::CompilationConnector
    void setOSRProhibitedOverRangeOfTrees() { _osrProhibitedOverRangeOfTrees = true; }
    bool isOSRProhibitedOverRangeOfTrees() { return _osrProhibitedOverRangeOfTrees; }
 
-   void setUseTracingBuffer(bool use) { _useTracingBuffer = use; }
-   bool useTracingBuffer() { return _useTracingBuffer; }
-   char *getTracingBufferStart() { return _tracingBufferStart; }
-   char *getTracingBufferCursor() { return _tracingBufferCursor; }
-   void incTracingBuffer(size_t size)
-      {
-      _tracingBufferCursor += size;
-      _tracingBufferFreeSpace -= size;
-      }
-   void allocateTracingBuffer();
-   void resetTracingBuffer()
-      {
-      _tracingBufferCursor = _tracingBufferStart;
-      _tracingBufferFreeSpace = _tracingBufferSize;
-      }
-   int32_t getTracingBufferFreeSpace() { return _tracingBufferFreeSpace; }
-
 private:
    enum CachedClassPointerId
       {
@@ -492,12 +459,6 @@ private:
 
    TR::SymbolValidationManager *_symbolValidationManager;
    bool _osrProhibitedOverRangeOfTrees;
-
-   bool _useTracingBuffer;
-   char *_tracingBufferStart;
-   char *_tracingBufferCursor;
-   int32_t _tracingBufferSize;
-   int32_t _tracingBufferFreeSpace;
    };
 
 }

--- a/runtime/compiler/compile/J9SymbolReferenceTable.cpp
+++ b/runtime/compiler/compile/J9SymbolReferenceTable.cpp
@@ -889,29 +889,21 @@ J9::SymbolReferenceTable::findOrCreateShadowSymbol(TR::ResolvedMethodSymbol * ow
       containingClass =
          owningMethod->definingClassFromCPFieldRef(comp(), cpIndex, isStatic, &fromResolvedJ9Method);
 
-      if (comp()->compileRelocatableCode())
-         {
-         TR_ASSERT_FATAL(
-            containingClass != NULL,
-            "failed to get defining class of field ref cpIndex=%d in owning method J9Method=%p\n"
-            "\tTR_ResolvedJ9Method::definingClassFromCPFieldRef=%p, TR_ResolvedRelocatableJ9Method::definingClassFromCPFieldRef=%p\n"
-            "\tfromResolvedJ9Method=%p\n"
-            "\tTR_UseSymbolValidationManager=%d",
-            cpIndex,
-            owningMethod->getNonPersistentIdentifier(),
-            owningMethod->TR_ResolvedJ9Method::definingClassFromCPFieldRef(comp(), owningMethod->cp(), cpIndex, isStatic),
-            static_cast<TR_ResolvedRelocatableJ9Method *>(owningMethod)->definingClassFromCPFieldRef(comp(), cpIndex, isStatic),
-            fromResolvedJ9Method,
-            comp()->getOption(TR_UseSymbolValidationManager));
-         }
-      else
-         {
-         TR_ASSERT_FATAL(
-            containingClass != NULL,
-            "failed to get defining class of field ref cpIndex=%d in owning method J9Method=%p",
+      // Normally, the expectation is that if a cp field ref is resolved, then the cp ref pointing
+      // to the containing class must also be resolved. However, in #9416, it was discovered that
+      // this assumption does not hold in some cases. Under AOT, loading class data from the SCC
+      // using findSharedData() may fail, which is propagated to the JIT as a NULL containing class.
+      // In addition, if a class is redefined between the point where a field is looked up and
+      // where the defining class is looked, the latter may become NULL.
+      //
+      // Having a NULL containing class would imply two fields from different classes could have
+      // the same lookup key, which is a problem. So, until the above issues are resolved, the
+      // safe thing to do is to abort compilation in these (rare) cases.
+      if (containingClass == NULL)
+         comp()->failCompilation<TR::CompilationException>(
+            "failed to get defining class of resolved field ref cpIndex=%d in owning method J9Method=%p",
             cpIndex,
             owningMethod->getNonPersistentIdentifier());
-         }
 
       ResolvedFieldShadowKey key(containingClass, offset, type);
       TR::SymbolReference *symRef =

--- a/runtime/compiler/compile/J9SymbolReferenceTable.cpp
+++ b/runtime/compiler/compile/J9SymbolReferenceTable.cpp
@@ -866,14 +866,6 @@ J9::SymbolReferenceTable::findOrCreateShadowSymbol(TR::ResolvedMethodSymbol * ow
    TR_ResolvedJ9Method * owningMethod =
       static_cast<TR_ResolvedJ9Method*>(owningMethodSymbol->getResolvedMethod());
 
-   if (comp()->compileRelocatableCode())
-      {
-      comp()->setUseTracingBuffer(true);
-      if (comp()->getTracingBufferStart() == NULL)
-         comp()->allocateTracingBuffer();
-      comp()->resetTracingBuffer();
-      }
-
    bool isVolatile = true, isFinal = false, isPrivate = false, isUnresolvedInCP;
    TR::DataType type = TR::NoType;
    uint32_t offset = 0;
@@ -899,22 +891,6 @@ J9::SymbolReferenceTable::findOrCreateShadowSymbol(TR::ResolvedMethodSymbol * ow
 
       if (comp()->compileRelocatableCode())
          {
-         if (comp()->useTracingBuffer())
-            {
-            if (containingClass == NULL)
-               {
-               const char *bufferPtr = comp()->getTracingBufferCursor();
-               for (char *cursor = comp()->getTracingBufferStart(); cursor < bufferPtr; cursor += TRACING_BUFFER_SEGMENT_SIZE)
-                  {
-                  J9VMThread *vmThread = comp()->fej9()->getCurrentVMThread();
-                  // each subsection *should* be NULL terminated
-                  fprintf(stderr, "%p: %s", vmThread, cursor);
-                  }
-               fprintf(stderr, "End (%d)\n", comp()->getTracingBufferFreeSpace());
-               }
-            comp()->setUseTracingBuffer(false);
-            }
-
          TR_ASSERT_FATAL(
             containingClass != NULL,
             "failed to get defining class of field ref cpIndex=%d in owning method J9Method=%p\n"
@@ -942,13 +918,8 @@ J9::SymbolReferenceTable::findOrCreateShadowSymbol(TR::ResolvedMethodSymbol * ow
          findResolvedFieldShadow(key, isVolatile, isPrivate, isFinal);
 
       if (symRef != NULL)
-         {
-         comp()->setUseTracingBuffer(false);
          return symRef;
-         }
       }
-
-   comp()->setUseTracingBuffer(false);
 
    TR::Symbol * sym = 0;
 

--- a/runtime/compiler/env/J9SharedCache.cpp
+++ b/runtime/compiler/env/J9SharedCache.cpp
@@ -867,12 +867,6 @@ TR_J9SharedCache::createClassKey(UDATA classOffsetInCache, char *key, uint32_t &
 UDATA *
 TR_J9SharedCache::rememberClass(J9Class *clazz, bool create)
    {
-   return rememberClass(create, NULL, clazz);
-   }
-
-UDATA *
-TR_J9SharedCache::rememberClass(bool create, TR::Compilation *comp, J9Class *clazz)
-   {
    UDATA *chainData = NULL;
 #if defined(J9VM_OPT_SHARED_CLASSES) && (defined(TR_HOST_X86) || defined(TR_HOST_POWER) || defined(TR_HOST_S390) || defined(TR_HOST_ARM) || defined(TR_HOST_ARM64))
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(fe());
@@ -885,10 +879,6 @@ TR_J9SharedCache::rememberClass(bool create, TR::Compilation *comp, J9Class *cla
    if (!isROMClassInSharedCache(romClass, &classOffsetInCache))
       {
       LOG(1,"\trom class not in shared cache, returning\n");
-      if (comp)
-         {
-         ADD_TRACING_BUFFER_MESSAGE(comp, "rememberClass (%p): rom class %p not in shared cache, returning\n", clazz, romClass);
-         }
       return NULL;
       }
 
@@ -898,14 +888,10 @@ TR_J9SharedCache::rememberClass(bool create, TR::Compilation *comp, J9Class *cla
 
    LOG(3, "\tkey created: %.*s\n", keyLength, key);
 
-   chainData = findChainForClass(clazz, key, keyLength, comp, romClass);
+   chainData = findChainForClass(clazz, key, keyLength);
    if (chainData != NULL)
       {
       LOG(1, "\tchain exists (%p) so nothing to store\n", chainData);
-      if (comp)
-         {
-         ADD_TRACING_BUFFER_MESSAGE(comp, "rememberClass: chain exists (%p) so nothing to store, %p, %p\n", chainData, clazz, romClass);
-         }
       return chainData;
       }
 
@@ -920,30 +906,18 @@ TR_J9SharedCache::rememberClass(bool create, TR::Compilation *comp, J9Class *cla
    if (chainLength > maxChainLength*sizeof(UDATA))
       {
       LOG(1, "\t\t > %d so bailing\n", maxChainLength);
-      if (comp)
-         {
-         ADD_TRACING_BUFFER_MESSAGE(comp, "rememberClass: chainLength larger than maxChainLength; bailing, %p %p\n", clazz, romClass);
-         }
       return NULL;
       }
 
    if (!fillInClassChain(clazz, chainData, chainLength, numSuperclasses, numInterfaces))
       {
       LOG(1, "\tfillInClassChain failed, bailing\n");
-      if (comp)
-         {
-         ADD_TRACING_BUFFER_MESSAGE(comp, "rememberClass: failed to fill in class chain; bailing, %p, %p\n", clazz, romClass);
-         }
       return NULL;
       }
 
    if (!create)
       {
-      LOG(1, "\tnot asked to create but could create, returning non-null, %p %p\n", clazz, romClass);
-      if (comp)
-         {
-         ADD_TRACING_BUFFER_MESSAGE(comp, "rememberClass: not asked to create but could create ; bailing, %p %p\n", clazz, romClass);
-         }
+      LOG(1, "\tnot asked to create but could create, returning non-null\n");
       return (UDATA *) 0x1;
       }
 
@@ -966,18 +940,10 @@ TR_J9SharedCache::rememberClass(bool create, TR::Compilation *comp, J9Class *cla
    if (chainData)
       {
       LOG(1, "\tstored data, chain at %p\n", chainData);
-      if (comp)
-         {
-         ADD_TRACING_BUFFER_MESSAGE(comp, "rememberClass: stored data, chain at %p, %p, %p\n", chainData, clazz, romClass);
-         }
       }
    else
       {
       LOG(1, "\tunable to store chain\n");
-      if (comp)
-         {
-         ADD_TRACING_BUFFER_MESSAGE(comp, "rememberClass: unable to store chain %p, %p\n", vmThread, romClass);
-         }
       TR::Options::getAOTCmdLineOptions()->setOption(TR_NoStoreAOT);
 
       setSharedCacheDisabledReason(SHARED_CACHE_CLASS_CHAIN_STORE_FAILED);
@@ -1123,7 +1089,7 @@ TR_J9SharedCache::romclassMatchesCachedVersion(J9ROMClass *romClass, UDATA * & c
    }
 
 UDATA *
-TR_J9SharedCache::findChainForClass(J9Class *clazz, const char *key, uint32_t keyLength, TR::Compilation *comp, J9ROMClass *romClass)
+TR_J9SharedCache::findChainForClass(J9Class *clazz, const char *key, uint32_t keyLength)
    {
    UDATA * chainForClass = NULL;
 #if defined(J9VM_OPT_SHARED_CLASSES) && (defined(TR_HOST_X86) || defined(TR_HOST_POWER) || defined(TR_HOST_S390) || defined(TR_HOST_ARM) || defined(TR_HOST_ARM64))
@@ -1143,12 +1109,6 @@ TR_J9SharedCache::findChainForClass(J9Class *clazz, const char *key, uint32_t ke
    //fprintf(stderr,"findChainForClass: key %.*s chain %p\n", keyLength, key, dataDescriptor.address);
    chainForClass = (UDATA *) dataDescriptor.address;
 #endif
-
-   if (comp)
-      {
-      ADD_TRACING_BUFFER_MESSAGE(comp, "findChainForClass %p, %p, %p, %p\n", vmThread, clazz, romClass, chainForClass);
-      }
-
    return chainForClass;
    }
 

--- a/runtime/compiler/env/J9SharedCache.hpp
+++ b/runtime/compiler/env/J9SharedCache.hpp
@@ -182,7 +182,6 @@ public:
       }
 
    virtual UDATA *rememberClass(J9Class *clazz, bool create=true);
-   virtual UDATA *rememberClass(bool create, TR::Compilation *comp, J9Class *clazz);
 
    virtual UDATA rememberDebugCounterName(const char *name);
    virtual const char *getDebugCounterName(UDATA offset);
@@ -393,7 +392,7 @@ private:
                          uint32_t numSuperclasses, uint32_t numInterfaces);
 
    bool romclassMatchesCachedVersion(J9ROMClass *romClass, UDATA * & chainPtr, UDATA *chainEnd);
-   UDATA *findChainForClass(J9Class *clazz, const char *key, uint32_t keyLength, TR::Compilation *comp=NULL, J9ROMClass *romClass=NULL);
+   UDATA *findChainForClass(J9Class *clazz, const char *key, uint32_t keyLength);
 
    /**
     * \brief Helper Method; Converts an offset into the ROMClass section into a pointer.

--- a/runtime/compiler/env/j9method.cpp
+++ b/runtime/compiler/env/j9method.cpp
@@ -1725,9 +1725,6 @@ TR_ResolvedRelocatableJ9Method::storeValidationRecordIfNecessary(TR::Compilation
       {
       if (aotStats)
          aotStats->numDefiningClassNotFound++;
-
-      ADD_TRACING_BUFFER_MESSAGE(comp, "relo::storeValRecordIfNec, definingClass NULL, %p, %d, %p\n", constantPool, cpIndex, ramMethod);
-
       return false;
       }
 
@@ -1739,10 +1736,7 @@ TR_ResolvedRelocatableJ9Method::storeValidationRecordIfNecessary(TR::Compilation
    // all kinds of validations may need to rely on the entire class chain, so make sure we can build one first
    classChain = fej9->sharedCache()->rememberClass(true, comp, definingClass);
    if (!classChain)
-      {
-      ADD_TRACING_BUFFER_MESSAGE(comp, "relo::storeValRecordIfNec, classChain NULL, %p, %p, %d, %p\n", definingClass, constantPool, cpIndex, ramMethod);
       return false;
-      }
 
    bool inLocalList = false;
    TR::list<TR::AOTClassInfo*>* aotClassInfo = comp->_aotClassInfo;
@@ -1774,7 +1768,6 @@ TR_ResolvedRelocatableJ9Method::storeValidationRecordIfNecessary(TR::Compilation
    if (inLocalList)
       {
       traceMsg(comp, "\tFound in local list, nothing to do\n");
-      ADD_TRACING_BUFFER_MESSAGE(comp, "relo::storeValRecordIfNec, Found in local list, nothing to do (%p, %d, %p)\n", constantPool, cpIndex, ramMethod);
       if (aotStats)
          {
          if (isStatic)
@@ -1798,13 +1791,10 @@ TR_ResolvedRelocatableJ9Method::storeValidationRecordIfNecessary(TR::Compilation
             aotStats->numNewCHEntriesInLocalList++;
          }
 
-      ADD_TRACING_BUFFER_MESSAGE(comp, "relo::storeValRecordIfNec, Created new AOT class info %p\n", classInfo);
-
       return true;
       }
 
    // should only be a native OOM that gets us here...
-   ADD_TRACING_BUFFER_MESSAGE(comp, "relo::storeValRecordIfNec, native OOM?\n");
    return false;
    }
 
@@ -1857,7 +1847,6 @@ TR_ResolvedRelocatableJ9Method::fieldAttributes(TR::Compilation * comp, int32_t 
                }
             else
                {
-               ADD_TRACING_BUFFER_MESSAGE(comp, "relo::fieldAttrib, %d, %p, %p\n", cpIndex, constantPool, ramMethod());
                fieldInfoCanBeUsed = storeValidationRecordIfNecessary(comp, constantPool, cpIndex, TR_ValidateInstanceField, ramMethod());
                }
             }
@@ -1910,8 +1899,6 @@ TR_ResolvedRelocatableJ9Method::fieldAttributes(TR::Compilation * comp, int32_t 
 #if defined(DEBUG_LOCAL_CLASS_OPT)
       unresolvedCtr++;
 #endif
-
-      ADD_TRACING_BUFFER_MESSAGE(comp, "relo::fieldAttrib, theFieldIsFromLocalClass will be false (%p, %d, %p)\n", constantPool, cpIndex, ramMethod());
       }
 
    if (unresolvedInCP)
@@ -2056,10 +2043,7 @@ TR_ResolvedRelocatableJ9Method::definingClassFromCPFieldRef(
       valid = storeValidationRecordIfNecessary(comp, cp(), cpIndex, isStatic ? TR_ValidateStaticField : TR_ValidateInstanceField, ramMethod());
 
    if (!valid)
-      {
-      ADD_TRACING_BUFFER_MESSAGE(comp, "relo::definingClassFromCPFieldRef !valid, %p, %d, %p\n", cp(), cpIndex, ramMethod());
       clazz = NULL;
-      }
 
    return clazz;
    }

--- a/runtime/compiler/env/j9method.cpp
+++ b/runtime/compiler/env/j9method.cpp
@@ -1734,7 +1734,7 @@ TR_ResolvedRelocatableJ9Method::storeValidationRecordIfNecessary(TR::Compilation
    void *classChain = NULL;
 
    // all kinds of validations may need to rely on the entire class chain, so make sure we can build one first
-   classChain = fej9->sharedCache()->rememberClass(true, comp, definingClass);
+   classChain = fej9->sharedCache()->rememberClass(definingClass);
    if (!classChain)
       return false;
 

